math/clamp.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+// This module is browser compatible.
+
+/**
+ * Clamp a number within the inclusive [min, max] range.
+ *
+ * @param num The number to be clamped
+ * @param limits The inclusive [min, max] range
+ * @returns The clamped number
+ *
+ * @example Usage
+ * ```ts
+ * import { clamp } from "@std/math/clamp";
+ * import { assertEquals } from "@std/assert";
+ * assertEquals(clamp(5, [1, 10]), 5);
+ * assertEquals(clamp(-5, [1, 10]), 1);
+ * assertEquals(clamp(15, [1, 10]), 10);
+ * ```
+ */
+export function clamp(num: number, limits: [min: number, max: number]): number {
+  const [min, max] = limits;
+  if (min > max) {
```

Comment:
Nit: Passing `limits` as an array means that this array needs to be allocated prior to passing it to this function. Given that math functions are often used in high performance scenarios it might be more desirable to avoid the need to allocate at all for determining whether a value is in-between two others.